lib/puppet/provider/package/pkg.rb

```diff
@@ -160,6 +160,13 @@ def insync?(is)
         warning(_("Implicit version %{should} has %{n} possible matches") % { should: should, n: n })
       end
       potential_matches.each { |p|
+
+        # If the version that we match is installed then we don't need
+        # to check if it's installed (or installable), just return true
+        if (is != :absent) && (p[:status] == 'installed')
+          return true
+        end
```

Comment:
nit, in ruby it's not needed to add parentheses and it's common to use a guard clause like:

```ruby
return true if is != :absent && p[:status] == 'installed'
```